### PR TITLE
Fix: Refine backward compatibility of OpenCL version

### PR DESCRIPTION
**Pull request by edsonke**
Hi @doe300,

I think the icd_loader.cpp can be more compatible via this commit, please help to review.

Thanks

Best regards

### CODE REVIEW

**Comment by doe300** [comment]
Is there a particular reason, why you want the functions to be disabled?
They should compile even with OpenCL in a maximum version of 1.2, dont't they?

**Comment by edsonke** [comment]
Hi,
I think it might not pass in the Raspbian Jessie but yes, the function ```clCreateProgramWithILKHR``` is in latest extension spec [OpenCL extension v1.2 rev25](https://www.khronos.org/registry/OpenCL/specs/opencl-1.2-extensions.pdf)

The thing is that Raspbian Jessie repository only provide old  OpenCL-Headers which doesn't contain the latest revision, and result in compile failed on Jessie.

So here is the proposals: (1) Add additional build flag for this API in cmake file, or (2) We just enable it and add ```#warn``` to leave some hint for others when they meet the issue?

Which is more preferred?

**Comment by doe300** [comment]
> So here is the proposals: (1) Add additional build flag for this API in cmake file, or (2) We just enable it and add #warn to leave some hint for others when they meet the issue?

(1) Since the officially supported OS/Toolset includes OpenCL headers with all the necessary function definitions, I will not conditionally add this function depending on whether the headers are current enough.
(2) I would be okay with issuing a warning with more information (and how to fix the error, e.g. a hint to update the OpenCL headers) when the compilation fails due to missing OpenCL function definitions. I just don't know how to find out if the headers are current, before the compiler throws an error in order to display the warning ...

About the SVM function, maybe we should just remove them completely from the ICD list, so noone calls them by mistake.

**Comment by edsonke** [comment]
Hi @doe300 
As your suggestion, I update the commit:
1. Check function (clCreateProgramWithIL) exist  in configuration level and throw warning if not exist to make hint
2. Remove those SVM functions in list
Best regards

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,6 +126,14 @@ if(BUILD_ICD)
 		message(STATUS "Found Khronos ICD Loader in version ${OCL_ICD_VERSION} in ${OCL_ICD_LIBDIR}")
 		include_directories(${OCL_ICD_INCLUDEDIR})
 	endif()
+
+	# Check clCreateProgramWithIL exist or not
+	set(CMAKE_REQUIRED_LIBRARIES OpenCL)
+	check_symbol_exists(clCreateProgramWithIL ${SYSROOT_CROSS}/usr/include/CL/cl.h AVAIL_clCreateProgramWithIL)
```
Comment (doe300) [inline CMakeLists.txt]: This is a neat trick:)

**Pull request merged**